SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDevAuthTokenGenerator.cpp

```diff
@@ -0,0 +1,99 @@
+
+#include "SpatialGDKDevAuthTokenGenerator.h"
+
+#include "Async/Async.h"
+#include "Framework/Notifications/NotificationManager.h"
+
+#include "SpatialCommandUtils.h"
+#include "SpatialGDKEditorSettings.h"
+#include "SpatialGDKSettings.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialGDKDevAuthTokenGenerator);
+
+FSpatialGDKDevAuthTokenGenerator::FSpatialGDKDevAuthTokenGenerator()
+	: bIsGenerating(false)
+{
+}
+
+void FSpatialGDKDevAuthTokenGenerator::UpdateSettings(FString DevAuthToken)
+{
+	USpatialGDKEditorSettings* GDKEditorSettings = GetMutableDefault<USpatialGDKEditorSettings>();
+	GDKEditorSettings->DevelopmentAuthenticationToken = DevAuthToken;
+	GDKEditorSettings->SaveConfig();
+	GDKEditorSettings->SetRuntimeDevelopmentAuthenticationToken();
+	GDKEditorSettings->SetRuntimeUseDevelopmentAuthenticationFlow();
+}
+
+void FSpatialGDKDevAuthTokenGenerator::DoGenerateDevAuthTokenTasks()
+{
+	bool bIsRunningInChina = GetDefault<USpatialGDKSettings>()->IsRunningInChina();
+	AsyncTask(ENamedThreads::AnyBackgroundThreadNormalTask, [this, bIsRunningInChina]
+	{
+		AsyncTask(ENamedThreads::GameThread, [this]()
+		{
+			this->ShowTaskStartedNotification(TEXT("Generating Development Authentication Token"));
+		});
+
+		FString DevAuthToken;
+		FString ErrorMessage;
+		if (SpatialCommandUtils::GenerateDevAuthToken(bIsRunningInChina, DevAuthToken, ErrorMessage))
+		{
+			AsyncTask(ENamedThreads::GameThread, [this, DevAuthToken]()
+			{
+				UpdateSettings(DevAuthToken);
+				this->ShowTaskEndedNotification(TEXT("Development Authentication Token Updated"), SNotificationItem::CS_Success);
+			});
+		}
+		else
+		{
+			UE_LOG(LogSpatialGDKDevAuthTokenGenerator, Error, TEXT("Failed to generate a development authentication token: %s"), *ErrorMessage);
+			AsyncTask(ENamedThreads::GameThread, [this]()
+			{
+				this->ShowTaskEndedNotification(TEXT("Failed to generate Development Authentication Token"), SNotificationItem::CS_Fail);
+			});
+		}
+	});
+}
+
+void FSpatialGDKDevAuthTokenGenerator::AsyncGenerateDevAuthToken()
+{
+	bool bExpected = false;
+	if (bIsGenerating.CompareExchange(bExpected, true))
+	{
+		DoGenerateDevAuthTokenTasks();
+	}
+	else
+	{
+		UE_LOG(LogSpatialGDKDevAuthTokenGenerator, Display, TEXT("A previous Development Authentication Token request is still pending. New request for generation ignored."));
+	}
+}
+
+void FSpatialGDKDevAuthTokenGenerator::ShowTaskStartedNotification(const FString& NotificationText)
+{
+	FNotificationInfo Info(FText::AsCultureInvariant(NotificationText));
+	Info.ExpireDuration = 5.0f;
+	Info.bFireAndForget = false;
+
+	TaskNotificationPtr = FSlateNotificationManager::Get().AddNotification(Info);
+
+	if (TaskNotificationPtr.IsValid())
+	{
+		TaskNotificationPtr.Pin()->SetCompletionState(SNotificationItem::CS_Pending);
+	}
+}
+
+void FSpatialGDKDevAuthTokenGenerator::ShowTaskEndedNotification(const FString& NotificationText, SNotificationItem::ECompletionState CompletionState)
```

Comment:
Nit: ShowTaskEndedNotification contains some functional code (bIsGenerating = false;)
It might be nice to rename the function to something like OnTaskEnded just so that people don't think the only purpose of the function is to show a notification.